internal/ingress/controller/parser/parser_test.go

```diff
@@ -1732,6 +1732,175 @@ func TestKongProcessClasslessIngress(t *testing.T) {
+			KongIngresses:    kongIngresses,
+			Services:         services,
+		})
+		assert.Nil(err)
+		state, err := Build(logrus.New(), store)
+		assert.Nil(err)
+		assert.NotNil(state)
+
+		assert.Equal(1, len(state.Services), "expected one knative service")
+		svc := state.Services[0]
+
+		assert.Equal(1, len(svc.Routes), "expected one route in knative service")
+		route := svc.Routes[0]
+
+		assert.Equal(kong.StringSlice("https"), route.Protocols, "expected https protocol after override")
+		assert.Equal(kong.Int(308), route.HTTPSRedirectStatusCode, "expected 308 after override")
+	})
+	t.Run("knative ingress without konghq.com/override", func(t *testing.T) {
+		ingresses := []*knative.Ingress{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "knative-ingress-without-override",
+					Namespace: "foo-ns",
+					Annotations: map[string]string{
+						"networking.knative.dev/ingress.class": annotations.DefaultIngressClass,
+					},
+				},
+				Spec: knative.IngressSpec{
+					Rules: []knative.IngressRule{
+						{
+							Hosts: []string{"my-func.example.com"},
+							HTTP: &knative.HTTPIngressRuleValue{
+								Paths: []knative.HTTPIngressPath{
+									{
+										Path: "/",
+										AppendHeaders: map[string]string{
+											"foo": "bar",
+										},
+										Splits: []knative.IngressBackendSplit{
+											{
+												IngressBackend: knative.IngressBackend{
+													ServiceNamespace: "foo-ns",
+													ServiceName:      "foo-svc",
+													ServicePort:      intstr.FromInt(42),
+												},
+												Percent: 100,
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		kongIngresses := []*configurationv1.KongIngress{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "https-only",
+					Namespace: "foo-ns",
+					Annotations: map[string]string{
+						"networking.knative.dev/ingress.class": annotations.DefaultIngressClass,
+					},
+				},
+				Route: &kong.Route{
+					Protocols:               kong.StringSlice("https"),
+					HTTPSRedirectStatusCode: kong.Int(308),
+				},
+			},
+		}
+		services := []*corev1.Service{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "foo-svc",
+					Namespace: "foo-ns",
+					Annotations: map[string]string{
+						"networking.knative.dev/ingress.class": annotations.DefaultIngressClass,
+					},
+				},
+			},
+		}
+		store, err := store.NewFakeStore(store.FakeObjects{
+			KnativeIngresses: ingresses,
+			KongIngresses:    kongIngresses,
+			Services:         services,
+		})
+		assert.Nil(err)
+		state, err := Build(logrus.New(), store)
+		assert.Nil(err)
+		assert.NotNil(state)
+
+		assert.Equal(1, len(state.Services), "expected one knative service")
+		svc := state.Services[0]
+
+		assert.Equal(1, len(svc.Routes), "expected one route in knative service")
+		route := svc.Routes[0]
+
+		assert.NotEqual(kong.StringSlice("https"), route.Protocols)
+		assert.Nil(route.HTTPSRedirectStatusCode)
+	})
```

Comment:
Please add in at least one test which ensure that other annotations (`konghq.com/protocols`, `konghq.com/strip-path` etc)work too. I expect such annotations to be the common case over `konghq.com/override`.

Reply:
added as requested.